services-to-main.sh

```diff
@@ -2,42 +2,34 @@
 
 set -e
 
-function get_upstream_branch {
-   local BRANCH_NAME=main
-   local exists=`git show-ref refs/heads/$BRANCH_NAME`
-   if [[ -z $exists ]]; then
-      BRANCH_NAME=main
-   fi;
-   echo "$BRANCH_NAME"
-}
-
-function get_upstream_name {
-   git ls-remote --exit-code upstream &>/dev/null || {
-      echo "origin"
-      return
-   }
-   echo "upstream"
-}
+REMOTE_NAME=
+BRANCH_NAME=
 
-function pull_latest_commit {
-   local BRANCH_NAME=$(get_upstream_branch)
-   local REMOTE_NAME=$(get_upstream_name)
+set_remote_branch() {
+  REMOTE_NAME=upstream
+  git ls-remote --exit-code "$REMOTE_NAME" &>/dev/null ||
+    REMOTE_NAME=origin
 
-   echo "git checkout $BRANCH_NAME && git pull $REMOTE_NAME $BRANCH_NAME ..."
-   git checkout -B $BRANCH_NAME;
-   git pull $REMOTE_NAME $BRANCH_NAME;
+  BRANCH_NAME=main
 }
 
-export -f pull_latest_commit
-export -f get_upstream_branch
-export -f get_upstream_name
+pull_latest_commit() {
+  echo "git checkout $BRANCH_NAME && git pull $REMOTE_NAME $BRANCH_NAME ..."
```

Comment:
We should go back to `git fetch $REMOTE_NAME` and `git checkout $REMOTE_NAME/$BRANCH_NAME`.
Otherwise you get an unclean main branch which is updated by pull but contains something custom commits.

Reply:
Which custom commits can this be?
I like the current state more, because then I can use it, too, to get the most recent state in every sub-repo.

Reply:
Maybe we can make use of command line flags to switch between `fetch & checkout` and `checkout & pull`.